include/substrait-mlir/Dialect/Substrait/IR/SubstraitTypes.td

```diff
@@ -180,14 +180,32 @@ def Substrait_AtomicTypes {
 /// One of the currently supported atomic types.
 def Substrait_AtomicType :
   AnyTypeOf<Substrait_SimpleTypes.types#Substrait_ParametrizedTypes.types>;
+  
+def Substrait_ListType : Substrait_Type<"List", "list"> {
+  let summary = "Substrait list type";
+  let description = [{
+    This type represents a list type.
+  }];
+  let parameters = (ins Substrait_AtomicType:$type);
```

Comment:
Same here: if the restriction to atomic types is intentional, then add a TODO. Adding support for arbitrary nesting of lists and structs is probably not trivial and it may be a good strategy to do it in two steps.